pkg/engine/task/task_apply.go

```diff
@@ -92,7 +94,8 @@ func apply(ro []runtime.Object, c client.Client) ([]runtime.Object, error) {
 // kubernetes native objects might be a problem because we cannot just compare the spec as the spec might have extra fields
 // and those extra fields are set by some kubernetes component
 // because of that for now we just try to apply the patch every time
-func patch(newObj runtime.Object, existingObj runtime.Object, c client.Client) error {
+// returns the object that has been applied to the cluster
+func patch(newObj runtime.Object, c client.Client) (runtime.Object, error) {
```

Comment:
we could have the same semantics as `client.Patch` of modifying the passed object.
```suggestion
func patch(newObj *runtime.Object, c client.Client) error {
```